guide/english/certifications/javascript-algorithms-and-data-structures/functional-programming/use-the-reduce-method-to-analyze-data/index.md

```diff
@@ -120,11 +120,19 @@ var watchList = [
 ];
 
 // Add your code below this line
-
-var averageRating = watchList.filter(x => x.Director === "Christopher Nolan").map(x => Number(x.imdbRating)).reduce((x1, x2) => x1 + x2) / watchList.filter(x => x.Director === "Christopher Nolan").length;
+var averageRating = 
+    watchList
+    // Use filter to find films directed by Christopher Nolan
+      .filter(film => film.Director === "Christopher Nolan")
+      // Use map to convert their ratings from strings to numbers
+      .map(film => Number(film.imdbRating))
+      // Use reduce to add together their ratings
+      .reduce((rating1, rating2) => rating1 + rating2) 
```

Comment:
While we are making parameter names more readable, the first parameter represents the accumulator so:
```suggestion
      .reduce((sumOfRatings, rating) => sumOfRatings + rating) 
```